### DISCUSSION TITLE
How to Initialize Chat Client in 0.7.0?

### DISCUSSION THREAD

**Discussion by MuckT** [WalletConnect/walletconnect-monorepo#2509] [Developer Support] [answered]
In `@walletconnect/chat-client` `0.7.0` it appears the chat init method has changed to require a `syncClient` and `SyncStoreController`; however, [the current docs](https://docs.walletconnect.com/2.0/web/chat/usage#initialize-your-walletconnect-chatclient-using-your-project-id) don't reflect this. I took a look at applying the changes similar to what I see in [chat-client-js/canary.spec.ts](https://github.com/WalletConnect/chat-client-js/blob/main/packages/chat-client/test/canary/canary.spec.ts#L49-L65), but was running into type errors. Any advice highly apperciated!

### Sample Code
```TypeScript
let chatClient: IChatClient | null = null
export async function getChatClient() {
  if (!chatClient) {
    const core = new Core({ projectId: process.env.REACT_APP_PROJECT_ID })
    const syncClient = await SyncClient.init({
      projectId: process.env.REACT_APP_PROJECT_ID,
      core,
    })
    chatClient = await ChatClient.init({
      logger: 'debug',
      keyserverUrl: process.env.REACT_APP_KEY_SERVER_URL || 'https://keys.walletconnect.com',
      projectId: process.env.REACT_APP_PROJECT_ID || '',
      relayUrl: 'wss://relay.walletconnect.com',
      core,
      syncClient,
      SyncStoreController: SyncStore
    })
  }
  return chatClient
}

```

Type Errors:
- `Type 'Core' is not assignable to type 'ICore'.`
- `Type 'SyncClient' is not assignable to type 'ISyncClient'.`
- `.../node_modules/@walletconnect/sync-client/dist/types/controllers/syncStore").SyncStore' is not assignable to type 'typeof import(".../node_modules/@walletconnect/chat-client/node_modules/@walletconnect/sync-client/dist/types/controllers/syncStore").SyncStore'.
  Types of construct signatures are incompatible.
`

**Comment by devceline** [comment, marked as answer]
Thank you for noting this, docs have been updated: https://docs.walletconnect.com/2.0/web/chat/usage
